✨ Add client layer to `business-on-behalf-rest-api`-bundle

daniel-rose opened the pull request:
- allow to set default company user by `RestBusinessOnBehalfRequestTransfer` over client (calls from glue or yves are possible)
- default flag will be set in `spy_company_user`
- there is only one default company user per customer
- add client layer to main autoload section in main composer.json
- ...

sourcecube commented on bundles/company-business-unit-order-budget/src/FondOfOryx/Zed/CompanyBusinessUnitOrderBudget/Business/Reducer/OrderBudgetReducer.php:
```diff
@@ -0,0 +1,67 @@
+            ->requireSubtotal()
+            ->getSubtotal();
+
+        $this->orderBudgetFacade->updateOrderBudget(
+            $orderBudgetTransfer->setBudget($currentBudget - $subTotal)
+        );
```
What happens if this is a negative result?

daniel-rose replied:
This case will be covered by the plugin "CompanyBusinessUnitOrderBudgetCheckoutPreConditionPlugin". It will be executed before checkout.

daniel-rose committed:
✨ Add client layer to `business-on-behalf-rest-api`-bundle
- allow to set default company user by `RestBusinessOnBehalfRequestTransfer` over client (calls from glue or yves are possible)
- default flag will be set in `spy_company_user`
- there is only one default company user per customer
- add client layer to main autoload section in main composer.json
- ...
